### PR TITLE
User profile revamp - Working on KB - 3734, 3739

### PR DESCRIPTION
**Pull request by vShyave**

### CODE REVIEW

File: library/ws-widget/collection/src/lib/_common/content-toc/confirmation-modal/confirmation-modal.component.html [hunk 1]
```diff
@@ -1,26 +0,0 @@
-<div class="modal-container">
-    <!-- Enroll content -->
-    <ng-container *ngIf="!data.enroll">
-        <div class="flex flex-col gap-4 text-center">
-            <div class="ws-mat-default-text font-bold">You’re one step away from enrolling!</div>
-            <div class="text-sm">This batch starting on 12/06/2023 - 23/06/2023, kindly go through the content and be
-                prepared.</div>
-            <div class="flex flex-col sm:flex-row gap-4">
-                <button class="flex justify-center cancel-button" (click)="handleCloseModal()">Cancel</button>
```
Comment (vishnubansaltarento) [inline library/ws-widget/collection/src/lib/_common/content-toc/confirmation-modal/confirmation-modal.component.html:9]: This PR should not have any code related to TOC Page, please check

File: library/ws-widget/collection/src/lib/_common/content-toc/enroll-modal/enroll-modal.component.scss [hunk 2]
```diff
@@ -1,40 +0,0 @@
...
-.modal-container {
-    padding: 24px;
-
-    .radio-outer-div {
-        padding: 8px 16px;
-        border-radius: 4px;
-        border: 1px solid rgba(0, 0, 0, 0.16);
-        opacity: 1;
-    }
-
```
Comment (vishnubansaltarento) [inline library/ws-widget/collection/src/lib/_common/content-toc/enroll-modal/enroll-modal.component.scss:12]: This PR should not have any code related to TOC page, please confirm

File: project/ws/app/src/lib/routes/app-toc/components/app-toc-home/app-toc-home.component.ts [hunk 3]
```diff
@@ -1688,30 +1686,6 @@ export class AppTocHomeComponent implements OnInit, OnDestroy, AfterViewChecked,
...
   }
 
-  handleEnrollBatch(): void {
-    const dialogRef = this.dialog.open(EnrollModalComponent, {
-      width: '420px',
-      data: { enroll: true },
-      panelClass: 'enroll-modal',
-      disableClose: true,
-    })
-
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/app-toc/components/app-toc-home/app-toc-home.component.ts:1698]: This PR should not have code changes related to TOC page, please confirm

File: project/ws/app/src/lib/routes/profile-v2/components/transfer-request/transfer-request.component.html [hunk 4]
```diff
@@ -0,0 +1,45 @@
+<div class="flex flex-col gap-4">
+    <div class="text-base font-semibold text-center">Select all the fields to make transfer request.</div>
+
+    <form class="flex flex-col gap-4" [formGroup]="transferRequestForm">
+        <div class="flex flex-col">
+            <div class="text-sm font-bold required">Organization</div>
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/profile-v2/components/transfer-request/transfer-request.component.html:6]: Hardcoded text should come from language files, where ever we are using like this, please update

File: project/ws/app/src/lib/routes/profile-v2/components/transfer-request/transfer-request.component.ts [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+      'name': this.transferRequestForm.value['organization'],
+      'designation': this.transferRequestForm.value['designation'],
+      'group': this.transferRequestForm.value['group'],
+    }
+    const postData: any = {
+      'request': {
+        'userId': this.configService.unMappedUser.id,
+        'employmentDetails': {
+          'departmentName': this.transferRequestForm.value['organization'],
+        },
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/profile-v2/components/transfer-request/transfer-request.component.ts:68]: please document, all these functions

File: project/ws/app/src/lib/routes/profile-v2/components/transfer-request/transfer-request.component.ts [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+    }
+    postData.request.profileDetails.professionalDetails.push(data)
+    this.userProfileService.editProfileDetails(postData)
+    .pipe(takeUntil(this.destroySubject$))
+    .subscribe((_res: any) => {
+      this.matSnackBar.open('Request sent successfully!')
+      this.handleCloseModal()
+    },         (error: HttpErrorResponse) => {
+      if (!error.ok) {
+        this.matSnackBar.open('Unable to do transfer request, please try again later!')
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/profile-v2/components/transfer-request/transfer-request.component.ts:82]: same as above for hardcoded text

File: project/ws/app/src/lib/routes/profile-v2/components/verify-otp/verify-otp.component.ts [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+  styleUrls: ['./verify-otp.component.scss'],
+})
+
+export class VerifyOtpComponent implements OnInit, OnDestroy {
+
+  private destroySubject$ = new Subject()
+  @ViewChild('timerDiv', { static: false }) timerDiv !: any
+  @Output() resendOTP = new EventEmitter<string>()
+  @Output() otpVerified = new EventEmitter<string>()
+  timeLeft = 90
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/profile-v2/components/verify-otp/verify-otp.component.ts:25]: hardcoded value should come from constant

File: project/ws/app/src/lib/routes/profile-v2/components/verify-otp/verify-otp.component.ts [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+
+  ngOnInit() {
+    this.startTimer()
+  }
+
+  startTimer() {
+    this.interval = setInterval(() => {
+      if (this.timeLeft > 0) {
+        this.timeLeft = this.timeLeft - 1
+        this.timerDiv.nativeElement.innerHTML = `${Math.floor(this.timeLeft / 60)}m: ${this.timeLeft % 60}s`
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/profile-v2/components/verify-otp/verify-otp.component.ts:44]: hardcoded value should come from constant

File: project/ws/app/src/lib/routes/profile-v2/components/verify-otp/verify-otp.component.ts [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+      } else {
+        clearInterval(this.interval)
+        this.showResendOTP = true
+      }
+    },                          1000)
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/profile-v2/components/verify-otp/verify-otp.component.ts:49]: same as above

File: project/ws/app/src/lib/routes/profile-v2/components/verify-otp/verify-otp.component.ts [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+    } else {
+      this.verifyMobileOTP()
+    }
+  }
+
+  verifyEmailOTP(): void {
+    this.otpService.verifyEmailOTP(this.otpEntered, this.data.value)
+    .pipe(takeUntil(this.destroySubject$))
+    .subscribe((_res: any) => {
+      this.matSnackbar.open("OTP verification done successfully!")
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/profile-v2/components/verify-otp/verify-otp.component.ts:64]: Hardcoded text should come from language files

File: project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.html [hunk 11]
```diff
@@ -1,4 +1,452 @@
...
+      </ng-container>
+      <div class="flex items-center justify-center profile-circle" *ngIf="!portalProfile?.profileImageUrl">
+        {{ nameInitials }}
+      </div>
+    </div>
+    <div class="flex-1 profile-info">
+      <div class="flex flex-col gap-2">
+        <div class="flex flex-row" *ngIf="!portalProfile?.professionalDetails || !portalProfile?.professionalDetails[0]?.verifiedKarmayogi">
+          <div class="profile-not-verified">
+            <div class="text">Profile not verified</div>
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.html]: hardcoded values should come from constant files

File: project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.html [hunk 12]
```diff
@@ -392,81 +764,6 @@
...
-                      }" mat-button (click)="filter('saved')" *ngIf="
-                        portalProfile.id === currentUser?.userId ||
-                        portalProfile.userId === currentUser?.userId
-                      ">
-                      {{ 'profileView.savedPosts' | translate }}
-                    </a>
-                  </div>
-                </div> -->
-        <!-- </mat-card> -->
-        <!-- </div>
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.html]: please remove unwanted code, if not required

File: project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.scss [hunk 13]
```diff
@@ -2,510 +2,308 @@
...
-  margin-left: $size-xs;
-  margin-top:#{$size-xs / 4}
+@mixin custom-mat-icon($font-size: 20px, $width: 20px, $height: 20px) {
+  font-size: $font-size;
+  width: $width;
+  height: $height; 
 }
 
-.container-balanced {
-  max-width: 1202px;
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.scss]: remove unwanted code, if not required

File: project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.ts [hunk 14]
```diff
@@ -1,40 +1,43 @@
...
-import { ConfigurationsService, ValueService } from '@sunbird-cb/utils';
-import { map } from 'rxjs/operators'
 import moment from 'moment'
+import { Subject } from 'rxjs'
+import { debounceTime, distinctUntilChanged, map, startWith, takeUntil } from 'rxjs/operators'
 
-import {
-  NsContent,
-  WidgetContentService,
-} from '@sunbird-cb/collection'
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.ts]: please remove unwanted code if not required and use regex from constants

File: project/ws/app/src/lib/routes/profile-v2/routes/profile/profile.component.html [hunk 15]
```diff
@@ -1,25 +1,9 @@
...
 
 <div *ngIf="currentRoute === 'home'" class="container-balanced-banner margin-top-s">
   <ng-container [wsResolverWidget]="banner"></ng-container>
 </div>
-<!--
-<mat-sidenav-container class="container-balanced" autosize name="discuss-menu">
-  <mat-sidenav [mode]="mode$ | async" [(opened)]="sideNavBarOpened" class="border-0">
-    <div class="contain">
-      <app-profile-v2-left-menu [unseen]="unread"></app-profile-v2-left-menu>
-    </div>
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/profile-v2/routes/profile/profile.component.html]: Remove unwanted code

File: src/app/competency-passbook/competency-card-details/competency-card-details.component.html [hunk 16]
```diff
@@ -110,13 +110,6 @@
                             </div>
                         </div>
                     </ng-container>
-                    <!-- <div class="flex items-center text-primary underline cursor-pointer text-xs" *ngIf="themeDetails.courseSubThemes[i]?.length > 2 && !eachCert.viewMore" (click)="handleViewMore(eachCert)">
-                        {{'competencyCardDetails.viewMore' | translate}}
-                    </div>
```
Comment (vishnubansaltarento) [inline src/app/competency-passbook/competency-card-details/competency-card-details.component.html]: same as above

File: library/ws-widget/collection/src/lib/_common/content-toc/confirmation-modal/confirmation-modal.component.html (hunk 1 above)
Comment (vShyave) [reply library/ws-widget/collection/src/lib/_common/content-toc/confirmation-modal/confirmation-modal.component.html:9]: Those are extra codes, which we are not at all using. Hence I cleaned up the code.

File: library/ws-widget/collection/src/lib/_common/content-toc/enroll-modal/enroll-modal.component.scss (hunk 2 above)
Comment (vShyave) [reply library/ws-widget/collection/src/lib/_common/content-toc/enroll-modal/enroll-modal.component.scss:12]: Same as above comment

File: project/ws/app/src/lib/routes/app-toc/components/app-toc-home/app-toc-home.component.ts (hunk 3 above)
Comment (vShyave) [reply project/ws/app/src/lib/routes/app-toc/components/app-toc-home/app-toc-home.component.ts:1698]: Same as above comment

File: project/ws/app/src/lib/routes/profile-v2/components/transfer-request/transfer-request.component.html (hunk 4 above)
Comment (vShyave) [reply project/ws/app/src/lib/routes/profile-v2/components/transfer-request/transfer-request.component.html:6]: I will do it everything at last, when I complete functionality

File: project/ws/app/src/lib/routes/profile-v2/components/transfer-request/transfer-request.component.ts (hunk 5 above)
Comment (vShyave) [reply project/ws/app/src/lib/routes/profile-v2/components/transfer-request/transfer-request.component.ts:68]: Let me close all the functionalities and then I will document each and every function

File: project/ws/app/src/lib/routes/profile-v2/components/transfer-request/transfer-request.component.ts (hunk 6 above)
Comment (vShyave) [reply project/ws/app/src/lib/routes/profile-v2/components/transfer-request/transfer-request.component.ts:82]: Will complete the functionality first, then I will work on these

File: project/ws/app/src/lib/routes/profile-v2/components/verify-otp/verify-otp.component.ts (hunk 7 above)
Comment (vShyave) [reply project/ws/app/src/lib/routes/profile-v2/components/verify-otp/verify-otp.component.ts:25]: Ok, but where that constant should be declared? In the same file or any other file. Please let me know.

File: project/ws/app/src/lib/routes/profile-v2/components/verify-otp/verify-otp.component.ts (hunk 8 above)
Comment (vShyave) [reply project/ws/app/src/lib/routes/profile-v2/components/verify-otp/verify-otp.component.ts:44]: same as above

File: project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.html (hunk 12 above)
Comment (vShyave) [reply project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.html]: Will remove once I implement all the functionality, as I need this code as backup or for reference

**Pull request merged**